### ISSUE TITLE
Stack overflow when compiling recursive input object types

### ISSUE THREAD

**Issue by justinmakaila** [apollographql/apollo-codegen#102]
There seems to be an issue in the codegen for this query:
```
query GetUserProjects($projectWhereArgs:ProjectWhereArgs!){
    viewer {
        allProjects(where:$projectWhereArgs) {
            edges {
                node {
                    id
                    title
                    description
                    users {
                        edges {
                            node {
                                id
                                username
                            }
                        }
                    }
                }
            }
        }
    }
}
```
It outputs this stack:
```
apollo-codegen generate **/*.graphql --schema schema.json --output API.swift
\RangeError: Maximum call stack size exceeded
    at getNamedType (/usr/local/lib/node_modules/apollo-codegen/node_modules/graphql/type/definition.js:148:25)
    at Compiler.addTypeUsed (/usr/local/lib/node_modules/apollo-codegen/src/compilation.js:102:26)
    at Compiler.addTypeUsed (/usr/local/lib/node_modules/apollo-codegen/src/compilation.js:102:14)
    at Compiler.addTypeUsed (/usr/local/lib/node_modules/apollo-codegen/src/compilation.js:102:14)
    at Compiler.addTypeUsed (/usr/local/lib/node_modules/apollo-codegen/src/compilation.js:102:14)
    at Compiler.addTypeUsed (/usr/local/lib/node_modules/apollo-codegen/src/compilation.js:102:14)
    at Compiler.addTypeUsed (/usr/local/lib/node_modules/apollo-codegen/src/compilation.js:102:14)
    at Compiler.addTypeUsed (/usr/local/lib/node_modules/apollo-codegen/src/compilation.js:102:14)
    at Compiler.addTypeUsed (/usr/local/lib/node_modules/apollo-codegen/src/compilation.js:102:14)
    at Compiler.addTypeUsed (/usr/local/lib/node_modules/apollo-codegen/src/compilation.js:102:14)
```
Without that query, the schema generates successfully.

Awaiting more input from the OP, this is just a proxy issue so it doesn't slip under the rug!

**Comment by FeltesPT** [comment]
Hey guys! OP here. I'm sharing the schema as requested.


[schema.json.zip](https://github.com/apollographql/apollo-codegen/files/965989/schema.json.zip)

**Comment by martijnwalraven** [comment]
Should be fixed in v0.10.12.
